sdk/storage/azblob/service/client.go

```diff
@@ -82,6 +83,32 @@ func NewClientFromConnectionString(connectionString string, options *ClientOptio
 	return NewClientWithNoCredential(parsed.ServiceURL, options)
 }
 
+// GetUserDelegationCredential obtains a UserDelegationKey object using the base ServiceURL object.
+// OAuth is required for this call, as well as any role that can delegate access to the storage account.
+func GetUserDelegationCredential(serviceURL string, ctx context.Context, info generated.KeyInfo, timeout *int32, requestID *string) (*UserDelegationCredential, error) {
```

Comment:
Why is this a free-standing func?  Seems like it should be a method on the `Client` type.